src/Entity/NetworkResponseEntity.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of Chrome PHP.
+ *
+ * (c) Soufiane Ghzal <sghzal@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace HeadlessChromium\Entity;
+
+/**
+ * HTTP response data.
+ *
+ * @see https://chromedevtools.github.io/devtools-protocol/tot/Network/#type-Response
+ */
+class NetworkResponseEntity
+{
+    /**
+     * Response URL.
+     *
+     * @var string
+     */
+    public $url;
+
+    /**
+     * HTTP response status code.
+     *
+     * @var int
+     */
+    public $status;
+
+    /**
+     * HTTP response status text.
+     *
+     * @var string
+     */
+    public $statusText;
+
+    /**
+     * HTTP response headers.
+     *
+     * @var array <string,string>
+     */
+    public $headers;
+
+    /**
+     * Resource mimeType as determined by the browser.
+     *
+     * @var string
+     */
+    public $mimeType;
+
+    public function __construct(
+        string $url,
+        int $status,
+        string $statusText,
+        array $headers,
+        string $mimeType
+    ) {
+        $this->url = $url;
+        $this->status = $status;
+        $this->statusText = $statusText;
+        $this->headers = $headers;
+        $this->mimeType = $mimeType;
+    }
+
+    /**
+     * Readonly trick for php < 8.1.
```

Comment:
I don't think this will work. __set only fires when there's not an actual property here. See https://3v4l.org/LVp0F. For this reason, I feel like either these should be replaced with private properties and getter methods, or alternatively, we should not bother with an entity class at all, and just use typed arrays.

```php
    /**
     * Network responses from all requests since the last navigation.
     *
     * @var array{url: string, status: int, statusText: string, headers: array<string, string>, mimeType: string}[]
     */
    private $networkResponses = [];
```

Reply:
How about using `__get` with private properties and add `@property-read` tags in the class docblock? https://3v4l.org/nFdbN

The types will be enforced, IDEs will show the properties in the auto complete feature, those properties will be read only and wel'll be able to replace everyting with the `readonly` keyword in php >=8.1 without changing the class behaviour.